tests/src/Interop/ICustomMarshaler/ICustomMarshaler.cs

```diff
@@ -0,0 +1,579 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.Runtime.InteropServices.Tests
+{
+    public class ICustomMarshalerTests : XunitBase
+    {
+        // To avoid having to create a native test library to reference in tests that
+        // interact with native libraries, we can use a simple method from the C standard
+        // library. Unfortunately, the C standard library has different names on Windows
+        // vs Unix.
+#if Windows
+        public const string LibcLibrary = "msvcrt.dll";
+#else
+        public const string LibcLibrary = "libc";
+#endif
+
+        [Fact]
+        public void CustomMarshaler_StringType_Success()
+        {
+            Assert.Equal(123, MarshalerOnStringTypeMethod("123"));
```

Comment:
Consider something like the following:
```
var val = "1234567";
var expected = Int32.Parse(val);
Assert.Equal(expected, MarshalerOnStringTypeMethod(val));
```
This is a less error prone way to author these kinds of tests.

Reply:
Alternatively you could start with an `int` and convert to a `string`. The point is to have the invariant data defined once.